pypesto/sample/sample.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import numpy as np
+
+from ..problem import Problem
+from ..result import Result
+from .sampler import Sampler
+from .adaptive_metropolis import AdaptiveMetropolisSampler
+
+logger = logging.getLogger(__name__)
+
+
+def sample(
+        problem: Problem,
+        n_samples: int,
+        sampler: Sampler = None,
+        x0: np.ndarray = None,
+        result: Result = None
+) -> Result:
+    """
+    This is the main function to call to do parameter sampling.
+
+    Parameters
+    ----------
+    problem:
+        The problem to be solved. If None is provided, a
+        :class:`pypesto.Pymc3Sampler` is used.
+    n_samples:
+        Number of samples to generate.
+    sampler:
+        The sampler to perform the actual sampling.
+    x0:
+        Initial parameter for the Markov chain. If None, the best parameter
+        found in optimization is used. Note that some samplers require an
+        initial parameter, some may ignore it.
+    result:
+        A result to write to. If None provided, one is created from the
+        problem.
+
+    Returns
+    -------
+    result:
+        A result with filled in sample_options part.
+    """
+    # prepare result object
+    if result is None:
+        result = Result(problem)
+
+    # try to find initial parameters
+    if x0 is None:
+        result.optimize_result.sort()
+        xs = result.optimize_result.get_for_key('x')
```

Comment:
how is this handled for parallel tempering?

Reply:
currently, all chains are initialized to the same starting point. of course, one could apply a clustering analysis and start from different starting points at some point, not sure about that ...

Reply:
I think it does not make sense to start all chains at the same point, then you lose the advantage of PT somehow.... How I usually do it is to look at the waterfall plot from the optimization and there I choose the different theta vectors for the initialization, i.e., different local optima.

Reply:
Sounds good. Not sure if there are theoretical results on starting PT chains from different points. I would assume that chains at high temperatures diverge fast anyway, but yes, mixing will be better if the starting points are different. To automatize, a clustering analysis will be necessary, which partly exists already in the visualization functions.

Reply:
I added now the possibility to manually define multiple starts. for automatic schemes I suggest to do that separately, in particular once priors are defined.